photon-lib/src/main/scala/com/linkedin/photon/ml/Types.scala

```diff
@@ -27,14 +27,17 @@ object Types {
   // A "sample" is a training, validation or scoring vector of feature values
   type UniqueSampleId = Long
 
-  // A CoordinateId is a fixed effect type, random effect type, or matrix factorization model type
+  // A CoordinateId corresponds to a fixed effect type, random effect type, or matrix factorization model type
   // (e.g. "per-user", "per-song" ...)
   type CoordinateId = String
 
   // RE stands for "random effect"
+  // REType is e.g. "per-song", "per-user"
   type REType = String
+  // REId is e.g. the song id, user id for which a random effect of the right type will be optimized
   type REId = String
 
-  // A "feature shard" is an arbitrary set of "feature bags"
+  // A "feature shard" is an arbitrary set of "feature bags", corresponding to a fixed effect type, random effect type,
+  // or matrix factorization model type
```

Comment:
That's not strictly true, since there isn't a 1-1 mapping between feature shards and coordinates - one shard could be used by multiple coordinates